plenum/test/view_change/test_prepare_in_queue_before_vc.py

```diff
@@ -0,0 +1,114 @@
+import functools
+from collections import deque
+
+import pytest
+
+from indy_node.server.node import logger
+from plenum.common.exceptions import SuspiciousNode
+from plenum.common.messages.node_messages import Prepare, Commit
+from plenum.common.util import compare_3PC_keys, max_3PC_key
+from plenum.server.models import Prepares
+from plenum.server.view_change.view_changer import ViewChanger
+from plenum.test.delayers import vcd_delay
+from plenum.test.helper import sdk_send_random_and_check
+from plenum.test.test_node import TestNode
+from plenum.test.view_change.helper import ensure_view_change
+
+REQ_COUNT = 10
+REQ_COUNT_AFTER_SLOW = 1
+
+
+@pytest.fixture(scope='module')
+def tconf(tconf):
+    old_max_3pc = tconf.Max3PCBatchSize
+    old_batch_wait = tconf.Max3PCBatchWait
+    tconf.Max3PCBatchSize = 1
+    tconf.Max3PCBatchWait = 1000
+    yield tconf
+
+    tconf.Max3PCBatchSize = old_max_3pc
+    tconf.Max3PCBatchWait = old_batch_wait
+
+
+def not_processing_prepare(node):
+    async def processNodeInBoxWithoutPrepare(self):
+        """
+        Process the messages in the node inbox asynchronously.
+        """
+        self.nodeIbStasher.process()
+        pprs = deque()
+        for i in range(len(self.nodeInBox)):
+            m = self.nodeInBox.popleft()
+            if isinstance(m, tuple) and len(
+                m) == 2 and not hasattr(m, '_field_types') and \
+                    (isinstance(m[0], Prepare) or isinstance(m[0], Commit)):
+                pprs.append(m)
+                continue
+            try:
+                await self.nodeMsgRouter.handle(m)
+            except SuspiciousNode as ex:
+                self.reportSuspiciousNodeEx(ex)
+                self.discard(m, ex, logger.debug)
+        self.nodeInBox.extend(pprs)
+    node.processNodeInBox = functools.partial(processNodeInBoxWithoutPrepare, node)
+
```

Comment:
Please add a test checking that we can finish view change if we got a quorum of view change done messages before we started the delayed view change.